src/NuGet.Clients/PackageManagement.VisualStudio/SolutionRestore/SolutionRestoreWorker.cs

```diff
@@ -53,7 +59,28 @@ internal sealed class SolutionRestoreWorker : ISolutionRestoreWorker, IDisposabl
             _serviceProvider = serviceProvider;
 
             _componentModel = _serviceProvider.GetService<SComponentModel, IComponentModel>();
+#if VS15
+            _solutionManager = _componentModel.GetService<IVsSolutionManager>();
 
+            if (_solutionManager == null)
+            {
+                throw new ArgumentNullException(nameof(_solutionManager));
+            }
+
+            IVsSolution vsSolution = serviceProvider.GetService(typeof(SVsSolution)) as IVsSolution;
+            if (vsSolution != null)
```

Comment:
there won't be a case where it will be null except for tests which is why i kept it as null check. But i'm open to throw as well.